src/v/cluster/metadata_dissemination_service.cc

```diff
@@ -156,21 +160,18 @@ ss::future<> metadata_dissemination_service::apply_leadership_notification(
     // the gate also needs to be taken on the destination core.
     return ss::with_gate(
       _bg, [this, ntp = std::move(ntp), lid, term]() mutable {
-          // the lock sequences the updates from raft
-          return _lock.with([this, ntp = std::move(ntp), lid, term]() mutable {
```

Comment:
don't you still need a mutex on core 0 here? if I understand correctly:

1. the lock on _every_ core enforces ordering at the source
2. a lock on core 0 prevents interleaved updates that may arrive from any core (because the work on core 0 isn't non-interruptible--its a fiber).

Reply:
reordering requests from different cores should be fine since they are coming from different raft groups